Added http cache

esrinivasan-rpx opened the pull request.

ethirajsrinivasan committed:
Added http cache

bragboy commented on lib/supercache.rb:
```diff
@@ -1,6 +1,7 @@
 require "supercache/version"
 require "supercache/engine"
 require "supercache/super_query_cache"
+require "supercache/super_request_cache"
```
`super_http_cache`

bragboy commented on lib/supercache/super_request_cache.rb:
```diff
@@ -0,0 +1,17 @@
+      end
+    end
+
+    alias_method :request_without_supercache, :request
```
`use Ruby 2.0 module / prepend`

ethirajsrinivasan committed:
Using prepend over alias_chain_method

ethirajsrinivasan committed:
Add Httpcache

*TODO replace alias_method with prepend when popular gems
 stop supporting ruby 1.9.3
